vault/router.go

```diff
@@ -611,3 +617,25 @@ func pathsToRadix(paths []string) *radix.Tree {
 
 	return tree
 }
+
+func filteredPassthroughHeaders(origHeaders map[string][]string, passthroughHeaders []string) map[string][]string {
+	retHeaders := make(map[string][]string)
+
+	// Create a map that uses lowercased header values as the key and the original
+	// header naming as the value for comparison down below.
+	lowerHeadersRef := make(map[string]string, len(origHeaders))
+	for key := range origHeaders {
```

Comment:
Is there a way we can avoid looping over the original headers? That could be expensive on every request.

Reply:
There not a really good way to access the values of the map without the exact key. We can mitigate this in most cases, since passthrough headers do not apply generally, by short circuiting this func.